staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_progress_test.go

```
@@ -0,0 +1,127 @@
+		wantRequestsSent++
+
+		PollConditionNoChange(pollPeriod, minimalNoChange, pollTimeout, func() bool {
+			requestsSent = pr.progressRequestsSentCount.Load()
+			return requestsSent == wantRequestsSent
+		})
+		if requestsSent != wantRequestsSent {
+			t.Errorf("Failed to wait for %d progress requests, got %d", wantRequestsSent, requestsSent)
+		}
+	}
+	pr.Remove()
+
+	logger.Info("No progress requests if no-one is waiting")
+	clock.Step(progressRequestPeriod * 2)
+	PollConditionNoChange(pollPeriod, minimalNoChange, pollTimeout, func() bool {
+		requestsSent = pr.progressRequestsSentCount.Load()
+		return requestsSent == wantRequestsSent
+	})
+	if requestsSent != wantRequestsSent {
+		t.Errorf("Failed to wait for %d progress requests, got %d", wantRequestsSent, requestsSent)
+	}
+
+	logger.Info("No progress after stopping")
+	close(stopCh)
+	pr.Add()
+	clock.Step(progressRequestPeriod * 2)
+	PollConditionNoChange(pollPeriod, minimalNoChange, pollTimeout, func() bool {
+		requestsSent = pr.progressRequestsSentCount.Load()
+		return requestsSent == wantRequestsSent
+	})
+	if requestsSent != wantRequestsSent {
+		t.Errorf("Failed to wait for %d progress requests, got %d", wantRequestsSent, requestsSent)
+	}
+}
+
+func newTestConditionalProgressRequester(clock clock.WithTicker) *testConditionalProgressRequester {
+	pr := &testConditionalProgressRequester{}
+	pr.conditionalProgressRequester = newConditionalProgressRequester(pr.RequestProgress, clock)
+	return pr
+}
+
+type testConditionalProgressRequester struct {
+	*conditionalProgressRequester
+	progressRequestsSentCount atomic.Int32
+}
+
+func (pr *testConditionalProgressRequester) RequestProgress(ctx context.Context) error {
+	pr.progressRequestsSentCount.Add(1)
+	return nil
+}
+
+func PollConditionNoChange(interval, stable, timeout time.Duration, condition func() bool) {
+	passCounter := 0
+	requiredNumberOfPasses := int(stable/interval) + 1
+	wait.Poll(interval, timeout, func() (done bool, err error) {
```

this is silently eating error - change to:
```
return wait.Poll(...
```

---

Done